### PR TITLE
feat(dlp): added a sample for inspect string omit overlap

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [inspect string omit overlap](https://cloud.google.com/dlp/docs/samples/dlp-inspect-string-omit-overlap) with unit test-case.

### CODE REVIEW

File: dlp/snippets/inspect/inspect_string_omit_overlap.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_omit_overlap.go:46]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_omit_overlap.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Specify the type of info the inspection will look for.
+	// See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types.
+	var infoTypes = []*dlppb.InfoType{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_omit_overlap.go:57]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_omit_overlap.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+		{Name: "PERSON_NAME"},
+		{Name: "EMAIL_ADDRESS"},
+	}
+
+	// Exclude EMAIL_ADDRESS matches
+	var exclusionRule = &dlppb.ExclusionRule{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_omit_overlap.go:63]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_omit_overlap.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+				},
+			},
+		},
+		MatchingType: dlppb.MatchingType_MATCHING_TYPE_PARTIAL_MATCH,
+	}
+
+	// Construct a ruleSet that applies the exclusion rule to the PERSON_NAME infoType.
+	// If a PERSON_NAME match overlaps with an EMAIL_ADDRESS match, the PERSON_NAME match will
+	// be excluded.
+	var ruleSet = &dlppb.InspectionRuleSet{
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_omit_overlap.go:77]: issue: remove `var`, use `:=`.

File: dlp/snippets/inspect/inspect_string_omit_overlap.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+			RuleSet: []*dlppb.InspectionRuleSet{
+				ruleSet,
+			},
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_string_omit_overlap.go:106]: issue: just return error to caller.

File: dlp/snippets/inspect/inspect_test.go [hunk 6]
```diff
@@ -276,3 +276,21 @@ func TestInspectBigquery(t *testing.T) {
 		})
 	}
 }
+
+func TestInspectStringOmitOverlap(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	buf := new(bytes.Buffer)
+
+	if err := inspectStringOmitOverlap(buf, tc.ProjectID, "gary@example.com"); err != nil {
+		t.Errorf("inspectStringOmitOverlap: %v", err)
```
Comment (telpirion) [inline dlp/snippets/inspect/inspect_test.go:285]: issue: replace `t.Errorf()` with `t.Fatal(err)`.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/inspect/inspect_string_omit_overlap.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_omit_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_omit_overlap.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_omit_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_string_omit_overlap.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_string_omit_overlap.go]: fixed it.

File: dlp/snippets/inspect/inspect_test.go (hunk 6 above)
Comment (AarshD-crest) [reply dlp/snippets/inspect/inspect_test.go]: fixed it.

**Review by telpirion — APPROVED** [review]

**Pull request merged**